pennylane/operation.py

```diff
@@ -651,19 +654,34 @@ def __matmul__(self, other):
 
         if isinstance(other, Observable):
             self.obs.append(other)
+
+            if QNode._current_context:
+                QNode._current_context.queue.remove(other)
+
             return self
 
         raise ValueError("Can only perform tensor products between observables.")
 
     def __rmatmul__(self, other):
         if isinstance(other, Observable):
             self.obs[:0] = [other]
+            
             return self
 
         raise ValueError("Can only perform tensor products between observables.")
 
     __imatmul__ = __matmul__
 
+    def queue(self):
+        """Append the operation to a QNode queue."""
```

Comment:
Here ``operation`` stands for ``observable``, right?

Reply:
Indeed!